plugins/operator-avg/src/main/scala/com/stratio/sparkta/plugin/operator/avg/AvgOperator.scala

```diff
@@ -31,15 +34,28 @@ class AvgOperator(properties: Map[String, JSerializable]) extends Operator(prope
 
   override val writeOperation = WriteOp.Avg
 
+  //scalastyle:off
   override def processMap(inputFields: Map[String, JSerializable]): Option[Number] = {
     if ((inputField.isDefined) && (inputFields.contains(inputField.get))) {
-      Some(inputFields.get(inputField.get).get.asInstanceOf[Number])
-    } else AvgOperator.SOME_ZERO_NUMBER
+      inputFields.get(inputField.get).get match {
+        case value if value.isInstanceOf[String] => Try(Some(value.asInstanceOf[String].toDouble.asInstanceOf[Number]))
+          .getOrElse(None)
```

Comment:
I think that you could create a function in operator which could be named like "getNumberFromSerializable" or something like that.